src/bots/HighSellVolumeBot.js

```diff
@@ -0,0 +1,255 @@
+    this._notifications = notifications
+    this._checkTimeInMilliseconds = checkTimeInMilliseconds
+
+    this._lastCheck = null
+    this._lastBuy = null
+    this._lastError = null
+  }
+
+  async _doStart () {
+    logger.debug({ msg: 'Initialized bot: ' + this.name })
+
+    this._markets.forEach(market => {
+      const sellToken = market.tokenA
+      const buyToken = market.tokenB
+      this._doRoutineLiquidityCheck(sellToken, buyToken)
+    })
+
+    // Check the liquidity periodically
+    setInterval(() => {
+      this._markets.forEach(market => {
+        const sellToken = market.tokenA
+        const buyToken = market.tokenB
+        this._doRoutineLiquidityCheck(sellToken, buyToken)
+      })
+    }, this._checkTimeInMilliseconds)
+  }
+
+  async _doStop () {
+    logger.debug({ msg: 'Bot stopped: ' + this.name })
+  }
+
+  async _doRoutineLiquidityCheck (sellToken, buyToken) {
+    // Do ensure liquidity on the market
+    auctionLogger.debug({
+      sellToken,
+      buyToken,
+      msg: "Doing a routine check. Let's see if we will be able to ensure the liquidity"
+    })
+    return this._checkBuyLiquidity({
+      sellToken,
+      buyToken,
+      from: this._botAddress
+    })
+  }
+
+  async _checkBuyLiquidity ({ sellToken, buyToken, from }) {
+    this._lastCheck = new Date()
+    let liquidityWasChecked
+    try {
+      // We must check both auction sides
+      await this._checkEnoughBalance({ sellToken, buyToken, from })
+      await this._checkEnoughBalance({ sellToken: buyToken, buyToken: sellToken, from })
+
+      liquidityWasChecked = true
+    } catch (error) {
+      this.lastError = new Date()
+      liquidityWasChecked = false
+      this._handleError(sellToken, buyToken, error)
+    }
+
+    return liquidityWasChecked
+  }
+
+  async _checkEnoughBalance ({ sellToken, buyToken, from }) {
+    logger.debug('Checking enough balance for %s-%s in %s', sellToken, buyToken, from)
+    const auctionSellVolumePromise = this._dxInfoService.getSellVolume({
+      sellToken, buyToken })
+    const accountBalancesPromise = this._dxInfoService.getBalances({
+      address: from })
+    const auctionIndexPromise = this._dxInfoService.getAuctionIndex({
+      sellToken, buyToken })
+    const externalPricePromise = this._marketService.getPrice({
+      tokenA: sellToken, tokenB: buyToken })
+    const [ auctionSellVolume, accountBalances, auctionIndex, externalPrice ] =
+    await Promise.all([
+      auctionSellVolumePromise,
+      accountBalancesPromise,
+      auctionIndexPromise,
+      externalPricePromise
+    ])
+
+    const lastAvailableClosingPrice = await this._dxInfoService.getLastClosingPrices({
+      sellToken, buyToken, auctionIndex })
+    const { amount: buyTokenBalance } = accountBalances.find(({ token, amount }) => {
+      if (token === buyToken) {
+        return amount
+      }
+    })
+    const estimatedPrice = lastAvailableClosingPrice[lastAvailableClosingPrice.length - 1]
```

Comment:
Why do we use the last closing price? is not more accurate to use the external feed only?

Reply:
I wasn't sure about how to handle this.

Maybe you're right and should be enough. Also it will simplify logic and code. We can discuss this point